tests/integration/simulacron/test_backpressure.py

```diff
@@ -0,0 +1,179 @@
+# Copyright DataStax, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import time
+
+from cassandra import OperationTimedOut
+from cassandra.cluster import Cluster, ExecutionProfile, EXEC_PROFILE_DEFAULT, NoHostAvailable
+from cassandra.policies import RoundRobinPolicy, WhiteListRoundRobinPolicy
+from tests.integration import requiressimulacron, libevtest
+from tests.integration.simulacron import SimulacronBase, PROTOCOL_VERSION
+from tests.integration.simulacron.utils import ResumeReads, PauseReads, prime_request, start_and_prime_singledc
+
+
+@requiressimulacron
+@libevtest
+class TCPBackpressureTests(SimulacronBase):
+    def setUp(self):
+        self.callback_successes = 0
+        self.callback_errors = 0
+
+    def callback_success(self, results):
+        self.callback_successes += 1
+
+    def callback_error(self, results):
+        self.callback_errors += 1
+
+    def _fill_buffers(self, session, query, expected_blocked=3, **execute_kwargs):
+        futures = []
+        buffer = '1' * 50000
+        for _ in range(100000):
+            future = session.execute_async(query, [buffer], **execute_kwargs)
+            futures.append(future)
+
+            total_blocked = 0
+            for pool in session.get_pools():
+                if not pool._connection._socket_writable:
+                    total_blocked += 1
+            if total_blocked == expected_blocked:
```

Comment:
is there any scenario where `total_blocked` can be greater than `expected_blocked`? Might be safer to just change it to `total_blocked >= expected_blocked`

Reply:
no chance since there is only 3 connections. But better to fix this yep.